File: routes/addressRoutes.js
// routes/addressRoutes.js
const express = require('express');
const router = express.Router();
const addressController = require('../controllers/addressController');

// Define routes
router.post('/', addressController.createAddress);       // Create new address
router.get('/', addressController.getAllAddresses);     // Get all addresses
router.get('/:id', addressController.getAddressById);   // Get single address by ID
router.put('/:id', addressController.updateAddress);    // Update address by ID
router.delete('/:id', addressController.deleteAddress); // Delete address by ID

module.exports = router;

File: routes/address_routes.py
from fastapi import APIRouter
from controllers.address_controller import AddressController  # Placeholder for your controller

router = APIRouter(prefix="/address", tags=["Address"])

@router.post("/", response_model=dict)
async def create_address(data: dict):
    return await AddressController.create_address(data)

@router.get("/", response_model=list)
async def get_all_addresses():
    return await AddressController.get_all_addresses()

@router.get("/{id}", response_model=dict)
async def get_address_by_id(id: str):
    return await AddressController.get_address_by_id(id)

@router.put("/{id}", response_model=dict)
async def update_address(id: str, data: dict):
    return await AddressController.update_address(id, data)

@router.delete("/{id}", response_model=dict)
async def delete_address(id: str):
    return await AddressController.delete_address(id)